### PR TITLE
Fan 1381 Testing load more button on home page

### PR DESCRIPTION
**Pull request by lizlux**
Adding two tests: 
* Is the fan feed displayed
* Is the load more button working with CORS

Fixed one test: 
* Feature image required

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+public class LoadMore extends BasePageObject {
+
+    @FindBy(css = ".more-stories .load-more-btn")
+    private WebElement loadMoreButton;
+
+    public boolean isButtonDisplayed() {
+        try {
+            return loadMoreButton.isDisplayed();
+        } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            return false;
+        }
+    }
+
+    public boolean areMorePostsDisplayed() {
+        try {
+            loadMoreButton.click();
+            WebElement element = wait.forElementPresent(By.cssSelector("[tracking-label='more-top-stories.item-15']"));
+            return element.isDisplayed();
+        } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/fandom/FanFeedTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.wikia.webdriver.testcases.fandom;
+
+import com.wikia.webdriver.elements.fandom.components.FanFeed;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/FanFeedTest.java]: MINOR Remove this unused import 'com.wikia.webdriver.elements.fandom.components.FanFeed'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java (hunk 1 above)
Comment (gabrys) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: Probably better define `waitForLoadMoreButton`

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java (hunk 2 above)
Comment (gabrys) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: Probably better define `waitForMorePosts`

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package com.wikia.webdriver.elements.fandom.components;
+
+import com.gargoylesoftware.htmlunit.ElementNotFoundException;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
+/**
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: Get rid of this Created by section

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java (hunk 2 above)
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: I would recommend splitting this into two, one method clicking and waiting for this `tracking-label='more- ...` (returning ) LoadMore and another method checking for visibility of posts returning boolean.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.wikia.webdriver.elements.fandom.components;
+
+import com.gargoylesoftware.htmlunit.ElementNotFoundException;
+import org.openqa.selenium.By;
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: If you want to get rid of sonar comments, just log info passing the exception object before returning false.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java (hunk 1 above)
Comment (lizlux) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: The button is already on the page so we don't have to wait for it here. I updated the name of the other function though.

**Comment by lizlux** [comment]
Hey @ludwikkazmierczak, let me know if the changes to the PR look okay to you. I didn't end up returning `LoadMore`, but rather the element that was waited for. It made more sense to me rather than re-selecting the same element. 

Please merge if it looks okay to you. Thanks!

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            PageObjectLogging.log("LoadMore", "Load More button is not displayed", true);
+            return false;
+        }
+    }
+
+    public boolean waitForMorePosts() {
+        try {
+            WebElement element = loadPosts();
+            return element.isDisplayed();
+        } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+public class LoadMore extends BasePageObject {
+
+    @FindBy(css = ".more-stories .load-more-btn")
+    private WebElement loadMoreButton;
+
+    public boolean isButtonDisplayed() {
+        try {
+            return loadMoreButton.isDisplayed();
+        } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+            PageObjectLogging.log("LoadMore", "Load More button is not displayed. " + e.getMessage(), false);
+            return false;
+        }
+    }
+
+    public boolean waitForMorePosts() {
+        try {
+            WebElement element = loadPosts();
+            return element.isDisplayed();
+        } catch (ElementNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**